docs/usage/advanced-hooks.md

```diff
@@ -40,19 +40,35 @@ import { renderHook, act } from '@testing-library/react-hooks'
 import { CounterStepProvider, useCounter } from './counter'
 
 test('should use custom step when incrementing', () => {
-  const wrapper = ({ children }) => <CounterStepProvider step={2}>{children}</CounterStepProvider>
-  const { result } = renderHook(() => useCounter(), { wrapper })
+  const wrapper = ({ children, step }) => (
+    <CounterStepProvider step={step}>{children}</CounterStepProvider>
+  )
+  const { result, rerender } = renderHook(() => useCounter(), {
+    wrapper,
+    initialProps: {
+      step: 2
+    }
+  })
 
   act(() => {
     result.current.increment()
   })
 
   expect(result.current.count).toBe(2)
+
+  rerender({ step: -2 })
```

Comment:
Given that this is a relatively niche feature, I think we would be better off have the basic example just with a `wrapper` and then some text about using `props` with the `wrapper` before showing the more complex example.

Thoughts?

Reply:
I moved the revision as a new section "Providing Props". It may help to show the use case about using `initialProps` and the `rerender` method with `wrapper`.